### PR TITLE
Batched GEMM for fp16

### PR DESCRIPTION
**Pull request by zjing14**
- Added a batched GEMM kernel
- Added instances for fp16 with all layouts

### CODE REVIEW

File: composable_kernel/include/utility/static_buffer_of_vector_type_v2.hpp [hunk 1]
```diff
@@ -104,6 +104,11 @@ struct StaticBufferOfVectorTypeV2 : public StaticallyIndexedArray<T, N>
             [&](auto i) { GetElement(i, true) = invalid_element_value_; });
     }
 
+    __host__ __device__ void Fill(VecBaseType v)
```
Comment (asroy) [inline composable_kernel/include/utility/static_buffer_of_vector_type_v2.hpp:107]: StaticBufferOfVectorTypeV2 should be deprecated. StaticBufferTupleOfVector should be used instead

File: host/host_tensor/include/host_gemm.hpp [hunk 2]
```diff
@@ -41,3 +41,45 @@ void host_gemm_mk_kn_mn(const Tensor<AType>& a_m_k,
...
                                c_m_n.mDesc.GetLengths()[1])(std::thread::hardware_concurrency());
 }
+
+template <typename AType,
+          typename BType,
+          typename CType,
+          typename AElementwiseOperation,
+          typename BElementwiseOperation,
+          typename CElementwiseOperation>
+void host_gemm_gmk_gkn_gmn(const Tensor<AType>& a_g_m_k,
```
Comment (asroy) [inline host/host_tensor/include/host_gemm.hpp:51]: Need to remove this, and add a reference operator like this one

https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/823657ed120144943b7db87c07fe3e647128db56/reference_operation/include/reference_gemm.hpp#L19

File: composable_kernel/include/utility/static_buffer_of_vector_type_v2.hpp (hunk 1 above)
Comment (zjing14) [reply composable_kernel/include/utility/static_buffer_of_vector_type_v2.hpp:107]: Done

Comment (zjing14) [reply composable_kernel/include/utility/static_buffer_of_vector_type_v2.hpp:107]: Done

File: host/host_tensor/include/host_gemm.hpp (hunk 2 above)
Comment (zjing14) [reply host/host_tensor/include/host_gemm.hpp]: Done

**Review by asroy — APPROVED** [review]

**Pull request merged**